model_analyzer/config/generate/optuna_run_config_generator.py

```diff
@@ -126,24 +145,70 @@ def get_configs(self) -> Generator[RunConfig, None, None]:
         # TODO: TMA-1885: Need an early exit strategy
         for _ in range(n_trials):
             trial = self._study.ask()
-            self._create_trial_objectives(trial)
-            run_config = self._create_objective_based_run_config()
+            trial_objectives = self._create_trial_objectives(trial)
+            run_config = self._create_objective_based_run_config(trial_objectives)
             yield run_config
             score = self._calculate_score()
             self._study.tell(trial, score)
 
-    def _create_trial_objectives(self, trial) -> None:
-        # TODO: TMA-1925: Use SearchParameters here
-        self._instance_count = trial.suggest_int("instance_count", 1, 8)
-        self._batch_size = int(2 ** trial.suggest_int("batch_size", 1, 10))
+    def _create_trial_objectives(self, trial: optuna.Trial) -> TrialObjectives:
+        trial_objectives: TrialObjectives = {}
+        for parameter_name in OptunaRunConfigGenerator.optuna_parameter_list:
+            parameter = self._search_parameters.get_parameter(parameter_name)
+
+            if parameter:
+                trial_objectives = self._add_trial_objective(
+                    trial, parameter_name, parameter, trial_objectives
+                )
 
         # TODO: TMA-1884: Need an option to choose btw. concurrency formula and optuna searching
-        self._concurrency = 2 * self._instance_count * self._batch_size
-        if self._concurrency > 1024:
-            self._concurrency = 1024
+        trial_objectives = self._set_objective_concurrency(trial_objectives)
+
+        return trial_objectives
+
+    def _add_trial_objective(
+        self,
+        trial: optuna.Trial,
+        name: str,
+        parameter: SearchParameter,
+        trial_objectives: TrialObjectives,
+    ) -> TrialObjectives:
+        if parameter.category is ParameterCategory.INTEGER:
+            objective = trial.suggest_int(
+                name, parameter.min_range, parameter.max_range
+            )
+        elif parameter.category is ParameterCategory.EXPONENTIAL:
+            objective = int(
+                2 ** trial.suggest_int(name, parameter.min_range, parameter.max_range)
+            )
+        elif parameter.category is ParameterCategory.INT_LIST:
+            objective = int(trial.suggest_categorical(name, parameter.enumerated_list))
+        elif parameter.category is ParameterCategory.STR_LIST:
+            objective = trial.suggest_categorical(name, parameter.enumerated_list)
+
+        trial_objectives[name] = objective
 
-    def _create_objective_based_run_config(self) -> RunConfig:
-        param_combo = self._create_parameter_combo()
+        return trial_objectives
+
+    def _set_objective_concurrency(
+        self, trial_objectives: TrialObjectives
+    ) -> TrialObjectives:
+        concurrency = (
```

Comment:
Is this going to work consistently? Or are we going to need to lean on Optuna to pick the load? I believe we've observed LLMs and other models not follow this rule.

Reply:
We are going to have a user option to either use the concurrency formula or have optuna pick the load (there's already a story filed for this).